cmd/zfs/zfs_main.c

```diff
@@ -2555,14 +2573,17 @@ print_us_node(boolean_t scripted, boolean_t parsable, int *fields, int types,
 		data_type_t type;
 		uint32_t val32;
 		uint64_t val64;
-		char *strval = NULL;
+		char *strval = "-";
 
 		while ((nvp = nvlist_next_nvpair(nvl, nvp)) != NULL) {
 			if (strcmp(nvpair_name(nvp),
 			    us_field_names[field]) == 0)
 				break;
 		}
 
+		if (nvp == NULL)
+			goto print;
```

Comment:
Rather than using a `goto` for this how about setting `type = DATA_TYPE_UNKNOWN` when `nvp == NULL` and adding appropriate cases to the switch statements to set `strval`.